pkg/server/public/handlers/database/create.go

```diff
@@ -24,6 +26,12 @@ func (h *Handler) Create() http.HandlerFunc {
 			return
 		}
 
+		if !ValidateDBStruct(&db) {
+			l.Error("invalid data")
+			utils.ErrorResponse(w, fmt.Errorf("invalid data"))
+			return
+		}
```

Comment:
We already have validation logic in the `pkg/api/types/validation/validation.go`
This should be 

```
if err := validation.ValidateDatabase(&db); err.HasAny(){
  utils.ErrorResponse(w, meta.NewInvalid(..., err)
  return
} 
```